source/core/DualTypeConversion.cpp

```diff
@@ -94,25 +94,24 @@ namespace Vireo
         return true;
     }
 
-    template<typename X, typename Y, typename Z> Y ConvertFromEnum(Z valueX, TypeRef typeRefY)
+    template<typename TSource, typename TDest> TDest ConvertFromEnum(TSource src, TypeRef destTypeRef)
     {
-        X integerValueX = static_cast<X>(valueX);
-        Y unsignedIntegerValueX = integerValueX >= 0 ? integerValueX : 0;
-        Y numElementsInY = static_cast<Y>(typeRefY->GetEnumItemCount());
-        if (unsignedIntegerValueX >= numElementsInY) {
-            unsignedIntegerValueX = numElementsInY - 1;
-        }
-        return unsignedIntegerValueX;
+        src = src >= 0 ? src : 0;
+        auto destEnumCount = static_cast<TDest>(destTypeRef->GetEnumItemCount());
+#pragma warning(push)
+#pragma warning(disable : 4018)  // Warning C4018 '>=': signed / unsigned mismatch
+        // It is safe to disable this warning because 'src' should be >= 0 here.
+        TDest dest = (src >= destEnumCount) ? destEnumCount - 1 : static_cast<TDest>(src);
+#pragma warning(pop)
+        return dest;
     }
 
-    template<typename T> void ApplyNumeric(T valueX, TypeRef typeRefY, void* pDataY)
+    template<typename T> void ApplyIntegralToNumeric(T valueX, TypeRef typeRefY, void* pDataY)
```

Comment:
'Integral' reads weird to me.  Why not just 'Integer'?

Reply:
I went with "Integral" to imply Integer and enum. Enum is sort of an integral type with underlying integer. If you feel strongly about it, I can fix this to make it 'Integer'.

Reply:
I don't feel that strongly, if you want to leave it, it's fine.

Reply:
I added some comments about the convention used in the file for Integer/Integral/Numeric.